spacelift/data_worker_pools.go

```diff
@@ -0,0 +1,78 @@
+package spacelift
+
+import (
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/pkg/errors"
+
+	"github.com/spacelift-io/terraform-provider-spacelift/spacelift/internal"
+	"github.com/spacelift-io/terraform-provider-spacelift/spacelift/internal/structs"
+)
+
+func dataWorkerPools() *schema.Resource {
+	return &schema.Resource{
+		Read: dataWorkerPoolsRead,
+		Schema: map[string]*schema.Schema{
+			"worker_pools": &schema.Schema{
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"worker_pool_id": {
+							Type:        schema.TypeString,
+							Description: "ID of the worker pool",
+							Required:    true,
+						},
+						"config": {
+							Type:        schema.TypeString,
+							Description: "credentials necessary to connect WorkerPool's workers to the control plane",
+							Computed:    true,
+							Sensitive:   true,
+						},
+						"name": {
+							Type:        schema.TypeString,
+							Description: "name of the worker pool",
+							Computed:    true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func dataWorkerPoolsRead(d *schema.ResourceData, meta interface{}) error {
+	var query struct {
+		WorkerPools []*structs.WorkerPool `graphql:"workerPools()"`
+	}
+	variables := map[string]interface{}{}
+
+	if err := meta.(*internal.Client).Query(&query, variables); err != nil {
+		return errors.Wrap(err, "could not query for worker pools")
+	}
+
+	workerPools := query.WorkerPools
+	if workerPools == nil {
+		return errors.New("worker pools not found")
+	}
+
+	wps := flattenDataWorkerPoolsList(workerPools)
+	if err := d.Set("worker_pools", wps); err != nil {
+		return errors.Wrap(err, "could not set worker pools")
+	}
+
+	return nil
+}
+
+func flattenDataWorkerPoolsList(workerPools []*structs.WorkerPool) []map[string]interface{} {
+	wps := make([]map[string]interface{}, 0, len(workerPools))
```

Comment:
Nit: if the length of the resulting slice is known, why not preallocate it directly, and do `wps[index]` instead of of `append`'ing?